src/Symfony/Bundle/FrameworkBundle/Console/Descriptor/JsonDescriptor.php

```diff
@@ -216,22 +216,23 @@ private function getContainerDefinitionData(Definition $definition, $omitTags =
             'synthetic' => $definition->isSynthetic(),
             'lazy' => $definition->isLazy(),
         );
+
         if (method_exists($definition, 'isSynchronized')) {
-            $data['synchronized'] = $definition->isSynchronized();
+            $data['synchronized'] = $definition->isSynchronized(false);
         }
 
         $data['abstract'] = $definition->isAbstract();
         $data['file'] = $definition->getFile();
 
-        if ($definition->getFactoryClass()) {
+        if ($definition->getFactoryClass(false)) {
             $data['factory_class'] = $definition->getFactoryClass();
```

Comment:
Well, if your own service uses the factoryClass, you already got the deprecation warning when loading your service definition. You don't need a second one here (which would be reported as being thrown by Symfony btw)